src/net.h

```diff
@@ -578,13 +573,13 @@ class V2Transport final : public Transport
 
     /** Lock for receiver-side fields. */
     mutable Mutex m_recv_mutex ACQUIRED_BEFORE(m_send_mutex);
-    /** In {GARBAUTH, VERSION, APP}, the decrypted packet length, if m_recv_buffer.size() >=
+    /** In {VERSION, APP}, the decrypted packet length, if m_recv_buffer.size() >=
      *  BIP324Cipher::LENGTH_LEN. Unspecified otherwise. */
     uint32_t m_recv_len GUARDED_BY(m_recv_mutex) {0};
     /** Receive buffer; meaning is determined by m_recv_state. */
     std::vector<uint8_t> m_recv_buffer GUARDED_BY(m_recv_mutex);
-    /** During GARBAUTH, the garbage received during GARB_GARBTERM. */
-    std::vector<uint8_t> m_recv_garbage GUARDED_BY(m_recv_mutex);
+    /** AAD expected in next received packet (currently used only for garbage). */
+    std::vector<uint8_t> m_recv_aad GUARDED_BY(m_recv_mutex);
```

Comment:
Conceptually it's a bit weird to receive an AAD in one message but then verify it in the next message. Perhaps it's cleaner to to keep `m_recv_garbage` around and explicitly move it into `m_recv_aad` as soon as we start receiving bytes for the decoy/version message?

That way we can `Assume` that `m_recv_aad` is empty when start receiving bytes and clear it when we're done processing a message. (this commit already does the latter)

Reply:
> Conceptually it's a bit weird to receive an AAD in one message but then verify it in the next message. Perhaps it's cleaner to to keep `m_recv_garbage` around and explicitly move it into `m_recv_aad` as soon as we start receiving bytes for the decoy/version message?

If I understand correctly, this is roughly equivalent to dropping the second commit again, and I don't think this was cleaner.

Reply:
> At the start of the version phase, the previous garbage will be in `m_recv_aad`.

This is the part that feels a bit strange to me. But may it's not too bad:

We could instead `Assume` that `m_recv_aad` is empty when we enter the `APP` and when we enter the `APP_READY` state. It can be cleared in the `VERSION` state and after each normal message in the `APP` state, which indeed this PR does.

Reply:
I suppose an `Assume` in `APP` may or may not be future proof, because it depends on how AADs will be used: do we know what to expect for the next message (as is the case here for garbage), do we infer it from the current message, is it some time dependent thing or even a constant?

Reply:
> I suppose an `Assume` in `APP` may or may not be future proof, because it depends on how AADs will be used: do we know what to expect for the next message (as is the case here for garbage), do we infer it from the current message, is it some time dependent thing or even a constant?

Well, if we ever have a v3 that uses AAD for more than the garbage, we could drop the `Assume`, so I guess that's not a big deal.

On the other hand, I don't see that much value in adding an `Assume`. The sender and receiver code are sufficiently independent. So unless we have the exact same bug in the sender logic, a bug in the receiver logic that leads to `m_recv_aad` being non-empty would lead to an immediate disconnect. That is, such a bug will certainly be noticed during testing, even without the `Assume`. (The `Assume` would just make it easier to find the cause.) 

My current thinking is that I don't want to touch the PR now that it has three ACKs. But I'll add an `Assume` if there's another good reason to touch it (or if you have strong opinions on this, of course.)

Reply:
Not totally relevant to the PR but some remarks:

> do we know what to expect for the next message (as is the case here for garbage)

I think that's true in 99.9% of the cases I can imagine.

> do we infer it from the current message,

For contents of the AEAD, that's excluded by design. It would just be possible for data in the "outer" header (currently just the length field). But even if we want some additional data there, this doesn't really contradict the current logic in the code (e.g., we could concatenate the current-message thing to whatever `m_recv_aad` is).

> is it some time dependent thing

That will be too fragile, i.e., we would abort the connection if clocks are off.

> even a constant?

Then there will be no reason to include it.

Reply:
Thanks for the clarifications. Agree that adding an `Assume` can wait.